doc/README.md

```diff
@@ -0,0 +1,421 @@
+and the apps themselves as device specific APK files as installed at time of backup.
+
+*Storage ID*: A storage ID is the SHA-256 hash of the content stored in the repository.
+This ID is required in order to load the file from the repository.
+
+## Repository Layout
+
+The name of all files in the repository starts with the lower case hexadecimal representation
+of the storage ID, which is the SHA-256 hash of the file's contents.
+This allows for easy verification of files for accidental modifications,
+like disk read errors, by simply running the program `sha256sum` on the file
+and comparing its output to the file name.
+
+All files in a repository are only written once and never modified afterwards.
+
+```console
+.SeedVaultAndroidBackup
+└── f35860ee961789fb5f92f467455acf165120a319e9dc27044282982111546f26
+    ├── 00
+    │   └── 001b527ebb5eb57f4934bafeb998cb08595ed7ced603d9d25bd3c50b338f939d
+    ├── 01
+    │   └── 01e61554a023c9c1053e026c8a70498fb4732c3ecaaad1bd44003185b493529b
+    ├── ...
+    ├── fe
+    │   └── fe94fd20382e76d0215a743f3f27879d9555947250504de5b0d45321f1f66c7a
+    ├── ff
+    │   └── ff2e9f9c75d211602c5a68f0471aa549e2499a3fa9496255b26678f4aad75a98
+    ├── 22a5af1bdc6e616f8a29579458c49627e01b32210d09adb288d1ecda7c5711ec.snapshot
+    ├── 3ec79977ef0cf5de7b08cd12b874cd0f62bbaf7f07f3497a5b1bbcc8cb39b1ce.snapshot
+```
+
+Historically, all data that Seedvault saves to backup storage
+is below a `.SeedVaultAndroidBackup` directory.
+It can contain one or more repositories as users may use the same backup storage
+for several devices or user accounts.
+As having to choose and remember a specific folder is considered bad UX
+for the regular Android user,
+Seedvault creates a repository for the user.
+
+The folder name is the ID of the repository.
+It is the result of applying HMAC-SHA256 with the "app backup repoId key"
+(see [cryptography](#cryptography)) to the
+[`ANDROID_ID`](https://developer.android.com/reference/android/provider/Settings.Secure#ANDROID_ID)
+which is a 64-bit number (expressed as a hexadecimal string) provided by the operating system.
+It is unique to each combination of app-signing key, user, and device.
+This design should guarantee that only one single app ever backs up to or prunes the repository.
+Also, if the user generates a new recovery key and thus all keys change,
+Seedvault will not attempt to back up into the repository tied to the old key.
+
+For restore, we iterate through all repositories and only offer snapshots for restore
+that can be decrypted with the key provided by the user.
+Hence, a restore may run on a second device while the first device is doing a backup.
+
+Note: A repository used for file backup is stored in a folder of the format `[ANDROID_ID].sv`
+and thus completely separate.
+
+## Data Format
+
+All files stored in the repository start with a version byte
+followed by an encrypted and authenticated payload (see also [Cryptography](#cryptography)).
```

Comment:
[Clarity] Later (line 271), it is written that only the backup version byte is authenticated, along with an explanation of why that is the case. Until reaching that line of this document, it sounds like all of the data is authenticated. I don't know how much it matters, but it might be helpful to clarify that earlier (like around here) somehow.

Reply:
The encrypted data is authenticated. What you refer to is the authenticated associated data (AAD). The version byte is used as AAD, because it would otherwise not be authenticated.

Reply:
I know I just responded to the other comment, so sorry for bouncing around, but I think I misinterpreted "otherwise not be authenticated" as meaning, "we wouldn't bother with authentication at all if not for wanting to authenticate the version byte" (implying, because it's not needed otherwise). Might be an uncommon misinterpretation. Rephrasing to something like, "to ensure it is also authenticated" would help.